pre_commit/languages/pygrep.py

```diff
@@ -45,6 +45,36 @@ def _process_filename_at_once(pattern: Pattern[bytes], filename: str) -> int:
     return retv
 
 
+def _process_filename_by_line_negated(
+        pattern: Pattern[bytes],
+        filename: str,
+) -> int:
+    retv = 1
+    with open(filename, 'rb') as f:
+        for line in f:
+            if pattern.search(line):
+                retv = 0
+                break
```

Comment:
this can be `return 0`

and then below you can
```python
else:
    output.write_line(filename)
    return 1
```

Reply:
Yes, much simpler, thanks!

Though here there wouldn't be an `else`, would there? It's necessary to check through all the lines before deciding to return 1

Reply:
I mean [`for: ... else: ...`](https://youtu.be/8P7lXLXR_3c?list=PLWBKAf81pmOaP9naRiNAqug6EBnkPakvY)

Reply:
Ah, I see, thanks! I see you have a video about this too, will watch now https://youtu.be/8P7lXLXR_3c